src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,200 @@
+                System.out.println("Not created");
+            }
+        }
+
+        try{
+            BufferedWriter writer = Files.newBufferedWriter(path);
+            for(int i = 0; i<numberOfTasks; i++){
+                writer.write(taskList.get(i).toSaveFormat());
+                if(!(i==numberOfTasks-1)){
+                    writer.newLine();
+                }
+            }
+            writer.close();
+        }catch (IOException e ){
+            System.out.println("Save not successful.");
+        }
+
+
+
+
+    }
+
+    private static void loadTasksFromTxt(){
+        String home = System.getProperty("user.home");
+        Path path = Paths.get(home, "Desktop", "IP", "ip", "data", "duke.txt");
+        boolean directoryExists = Files.exists(path);
+        if (directoryExists){
+            File dataFile = new File(path.toString()); // create a File for the given file path
+            try{
+                Scanner sc = new Scanner(dataFile);
+                while (sc.hasNext()) {
+                    taskList.add(Task.fromSaveFormatString(sc.nextLine()));
+                }
+            }catch (FileNotFoundException e){
+                System.out.println("No data file found, existing task list is empty.");
+            }
+
+        }
+    }
+
+
+    private enum MainCommand{
+        LIST, BYE, TODO, EVENT, DEADLINE, DONE, DELETE
+    }
+
+    public static void deleteTask(int index){
+        Task removedTask = taskList.remove(index-1);
+        System.out.println("Noted. I've removed this task:\n" + removedTask.toString() +"\nNow you have "+ taskList.size() + " tasks in the list.");
+    }
+
+    public static void printCommandList(){
+        int i;
+        for (i=0; i< taskList.size(); i++){
+            if (taskList.get(i).isCompleted()) {
+                System.out.println((i+1) + ". " +taskList.get(i)); // use a tostring method
+            } else{
+                System.out.println((i+1) + ". " +taskList.get(i));
+            }
+        }
+    }
+
+    public static MainCommand findMainCommand(String command){
+        if (command.toLowerCase().matches("^deadline.*$")) {
+            return MainCommand.DEADLINE;
+        } else if (command.toLowerCase().matches("^todo.*$")){
+            return MainCommand.TODO;
+        } else if (command.toLowerCase().matches("^event.*$")){
+            return MainCommand.EVENT;
+        } else if (command.equals("bye")){
+            return MainCommand.BYE;
+        } else if (command.toLowerCase().equals("list")) {
+            return MainCommand.LIST;
+        } else if (command.toLowerCase().matches("done [1-9]([0-9]{2})?")) {
+            return MainCommand.DONE;
+        } else if (command.toLowerCase().matches("delete [1-9]([0-9]{2})?")){
+            return MainCommand.DELETE;
+        } else {
+            return null;
+        }
+    }
+
+    ////// COULD WRITE A FUNCTION TO EXTRACT THE COMMAND ///////
+
+    public static void main(String[] args) {
```

Comment:
Try to avoid long methods that is greater than 30 lines. Could refer to some common refactoring tips to make methods shorter